### PR TITLE
don't create home directory and missing parents when create_home == false

### PR DESCRIPTION
**Pull request by bmillemathias**
##### SUMMARY
related to https://github.com/ansible/ansible/issues/70589

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
user

### CODE REVIEW

**Comment by bmillemathias** [comment]
Actually I don't what would be the correct fix
- create the parent directory by not the home directory
- create nothing.

File: lib/ansible/modules/user.py [hunk 1]
```diff
@@ -634,11 +634,11 @@ def create_user_useradd(self):
 
         if self.home is not None:
             # If the specified path to the user home contains parent directories that
-            # do not exist, first create the home directory since useradd cannot
+            # does not exist, first create the home directory since useradd cannot
             # create parent directories
             parent = os.path.dirname(self.home)
             if not os.path.isdir(parent):
-                self.create_homedir(self.home)
+                os.makedirs(parent)
```
Comment (samdoran) [inline lib/ansible/modules/user.py]: Creating dirs recursively has a number of security problems and we generally avoid this.

Comment (bmillemathias) [reply lib/ansible/modules/user.py]: ok, there is nothing about that in python doc, Thanks

**Comment by bmillemathias** [comment]
Hi @samdoran 

Everything clean up, ready for review.

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
Please create a [backport PR](https://docs.ansible.com/ansible/latest/community/development_process.html#backporting-merged-prs) against `stable-2.9` and `stable-2.10` for this to be included in previous versions.

**Comment by bmillemathias** [comment]
> 
> 
> Please create a [backport PR](https://docs.ansible.com/ansible/latest/community/development_process.html#backporting-merged-prs) against `stable-2.9` and `stable-2.10` for this to be included in previous versions.

Done.

**Pull request merged**